Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -2007,6 +2007,11 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
 WebExtensionContext::TabVector WebExtensionContext::openTabs(IgnoreExtensionAccess ignoreExtensionAccess) const
 {
     return WTF::compactMap(m_tabMap, [&](auto& entry) -> std::optional<Ref<WebExtensionTab>> {
+        if (!entry.value->isOpen()) {
+            ASSERT_NOT_REACHED("The tab should be open at this point; please file a bug.");
```

Comment:
This might still fire in normal cases, if an app call things like `didChangeTabProperties:` before `didOpenTab:`. It would not be a bug.

The logging I think we should add is in `WebExtensionContext::getOrCreateTab`. That will be the stack trace the app bug is happening.

If a `WebExtensionTab` is created or gotten that has `webViews()` with `webView.configuration._webExtensionController != extensionContext()->extensionController()->wrapper()` then we will know what code in the app is causing it (instead of finding out later when `webViews()` is used in unrelated code.) That is the better place to have the `THROW_UNLESS` instead of in `webViews()`.

Reply:
Instead of `THROW_UNLESS` we can use `RELEASE_LOG_FAULT` to get a stack trace (simulated crash) without crashing. That will be actionable if done in `getOrCreateTab()`.